src/elements/primitives.rs

```diff
@@ -793,6 +799,25 @@ mod tests {
 		);
 	}
 
+	#[test]
+	#[should_panic]
+	fn varint7_invalid() {
+		deserialize_buffer::<VarInt7>(&[240]).expect("fail");
```

Comment:
What do you think about changing `#[should_panic]` + `expect` to the explicit check
like

```rust
match deserialize_buffer::<VarInt7>(&[240]) {
    Err(Error::InvalidVarInt7(_)) => {},
    _ => panic!()
}
```